src/libslic3r/ExtrusionEntity.hpp

```diff
@@ -160,10 +160,14 @@ class ExtrusionPath : public ExtrusionEntity
     // Height of the extrusion, used for visualization purposes.
     float height;
 
-    ExtrusionPath() : mm3_per_mm(-1), width(-1), height(-1), m_role(erNone), m_no_extrusion(false) {}
-    ExtrusionPath(ExtrusionRole role) : mm3_per_mm(-1), width(-1), height(-1), m_role(role), m_no_extrusion(false) {}
-    ExtrusionPath(ExtrusionRole role, double mm3_per_mm, float width, float height, bool no_extrusion = false) : mm3_per_mm(mm3_per_mm), width(width), height(height), m_role(role), m_no_extrusion(no_extrusion) {}
-    ExtrusionPath(double overhang_degree, int curve_degree, ExtrusionRole role, double mm3_per_mm, float width, float height) : overhang_degree(overhang_degree), curve_degree(curve_degree), mm3_per_mm(mm3_per_mm), width(width), height(height), m_role(role) {}
+    //These properties are just for staggered perimeter production.
+    float z_offset; //z_offset to be multiplied to the layer height, default is 0
```

Comment:
Do you mind to rename it to something like `staggered_perimeter_z_offset` to reduce confusion?